composables/api/useUserData.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from 'vue-query';
+import type { Database } from '~/types/generated-types';
+
+export const useUserData = () => {
+  const client = useSupabaseClient<Database>();
+  const user = useSupabaseUser();
+
+  const result = useQuery(['userData', user], async () => {
```

Comment:
It runs after `user` is changed, so always when we log in or log out.

Reply:
@zakrzaq I would need your help with it, to show me how to generate such types 🙏

Reply:
Anytime